Bottom: 6ad097cd18cce6245a23fcd6d1a92cbbcbc45b88
Top:    fa7d9fbd56a739bae7cebd723e6c59fcaa0466e9
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-09-21 11:49:50 -0500

Enable file-based secrets provider


---
diff --git a/roles/buildbot-master/files/master.cfg b/roles/buildbot-master/files/master.cfg
index 11942aaa9..c453a09b1 100644
--- a/roles/buildbot-master/files/master.cfg
+++ b/roles/buildbot-master/files/master.cfg
@@ -5,7 +5,7 @@
 # 'master.cfg' in your buildmaster's base directory.
 
 import os
-from buildbot.plugins import steps, util
+from buildbot.plugins import secrets, steps, util
 
 os.environ['LONG_RUN_TIMEOUT'] = '14400'
 
@@ -22,6 +22,9 @@ c['change_source'] = []
 # --master option)
 c['protocols'] = {"pb": {"port": int(os.getenv('BUILDBOT_WORKER_PORT', 9989))}}
 
+# SECRETS MANAGEMENT
+c['secretsProviders'] = [secrets.SecretInAFile(dirname="/run/secrets")]
+
 # STATUS TARGETS
 
 # 'status' is a list of Status Targets. The results of each build will be
diff --git a/roles/buildbot-worker-ntel/tasks/main.yml b/roles/buildbot-worker-ntel/tasks/main.yml
index c6c571016..7c865e5cc 100644
--- a/roles/buildbot-worker-ntel/tasks/main.yml
+++ b/roles/buildbot-worker-ntel/tasks/main.yml
@@ -79,6 +79,8 @@
   copy:
     content: "{{ known_hosts }}\n"
     dest: '{{ secrets_hostdir }}/{{ secrets_known_hosts }}'
+    group: '{{ buildbot_uid }}'
+    owner: '{{ buildbot_uid }}'
 
 - name: Checkout image repo
   git:
diff --git a/roles/buildbot-worker-ptxdist/tasks/main.yml b/roles/buildbot-worker-ptxdist/tasks/main.yml
index 46bf6355e..c3a635458 100644
--- a/roles/buildbot-worker-ptxdist/tasks/main.yml
+++ b/roles/buildbot-worker-ptxdist/tasks/main.yml
@@ -86,6 +86,8 @@
   copy:
     content: "{{ known_hosts }}\n"
     dest: '{{ secrets_hostdir }}/{{ secrets_known_hosts }}'
+    group: '{{ buildbot_uid }}'
+    owner: '{{ buildbot_uid }}'
 
 - name: Checkout image repo
   git:
